app/src/main/java/org/simple/clinic/patient/SimpleVideoModule.kt

```diff
@@ -2,27 +2,45 @@ package org.simple.clinic.patient
 
 import com.f2prateek.rx.preferences2.Preference
 import com.f2prateek.rx.preferences2.RxSharedPreferences
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
 import dagger.Module
 import dagger.Provides
+import org.simple.clinic.patient.SimpleVideo.Type.NumberOfPatientsRegistered
+import org.simple.clinic.patient.SimpleVideo.Type.TrainingVideoYoutubeId
+import org.simple.clinic.remoteconfig.RemoteConfigService
 import java.util.Locale
-import javax.inject.Named
 
 @Module
 class SimpleVideoModule {
 
   @Provides
-  @Named("number_of_patients_registered")
+  @SimpleVideo(NumberOfPatientsRegistered)
   fun provideCountOfRegisteredPatients(rxSharedPreferences: RxSharedPreferences): Preference<Int> {
     return rxSharedPreferences.getInteger("number_of_patients_registered", 0)
   }
 
   @Provides
-  @Named("training_video_youtube_id")
-  fun provideSimpleVideoUrlBasedOnLocale(locale: Locale): String {
-    return when (locale.language) {
-      "hi" -> "nHsQ06tiLzw"
-      // Default to English
-      else -> "YO3D1paAuqU"
-    }
+  @SimpleVideo(TrainingVideoYoutubeId)
+  fun provideSimpleVideoUrlBasedOnLocale(
+      uuidSetJsonAdapter: JsonAdapter<Set<Languages>>,
+      remoteConfigService: RemoteConfigService,
```

Comment:
We can also inject the remote config reader directly instead of injecting the service if the only thing we need is the reader.